src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -88,8 +88,10 @@ public ProCombatMoveAI(final ProAI ai) {
       clearedTerritories.add(patd.getTerritory());
     }
     territoryManager.populateEnemyAttackOptions(clearedTerritories, new ArrayList<>());
+    Set<Territory> territoriesToCheck = new HashSet<>(clearedTerritories);
+    territoriesToCheck.addAll(ProData.myUnitTerritories);
     Map<Territory, Double> territoryValueMap =
-        ProTerritoryValueUtils.findTerritoryValues(player, new ArrayList<>(), clearedTerritories);
+        ProTerritoryValueUtils.findTerritoryValues(player, new ArrayList<>(), clearedTerritories, territoriesToCheck);
```

Comment:
As a thought, a `TerritoryValueCalculator` object could encapsulate some of the calculations, and cut down on the parameters, and perhaps be a good place to tuck away some of the complexity from `ProCombatMoveAi`